R/rt_search.R

```diff
@@ -24,20 +35,19 @@
 #' }
 
 rt_search <- function(query, orderBy = NULL, format="l", rt_base = getOption("rt_base")) {
-  base_api <- paste(stringr::str_replace(rt_base, "\\/$", ""), # removes trailing slash from base URL just in case
-                    "REST", "1.0", sep = "/")
+  base_api <- rt_url(rt_base, "search/ticket?")
 
-  url <- paste0(base_api, "/search/ticket?query=", query)
+  #based on httr::modify_url()
+  #possible TODO - turn this into its own function that can be used internally in the package
+  l <- plyr::compact(list(query = query,
```

Comment:
Cool, hadn't seen plyr::compact. Do you think it's possible to do this without adding another dependency? Can `tidyr` do this so we don't need another dep?

Reply:
I can't seem to find it in dplyr/tidyr version of it, though I've noticed people sometimes include the code of other functions in a package to avoid exporting it (I've seen this in particular for %>%).  It seems like this could lead to some naming conflicts if both are loaded, but perhaps that's okay in this case?  

I asked Twitter and will maybe get an answer there...

Reply:
> It seems like this could lead to some naming conflicts if both are loaded, but perhaps that's okay in this case?

I'm not totally sure if you would so long as you don't export the binding.


The code for `plyr::compact` is just: `function(l) Filter(Negate(is.null), l)`

so it'd (1) easy to re-write in `purrr` or just copy the body of the function in and add a comment like `# In-lined from https://github.com/hadley/plyr/blob/fe192412748d1ccb6d524caeb5bcf2b5c474fd46/R/utils.r#L42`

Reply:
Ok, so apparently `purrr::compact()` exists, so I'll just use that (it's listed with `keep` in the documentation)